From cbdb44b216ded4c260fe7d518145507e3753e91e Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:10 -0400
Subject: [powerpc] boot: Add PROM_ERROR define in oflib

Message-id: <1410545655-205645-502-git-send-email-dzickus@redhat.com>
Patchwork-id: 94425
O-Subject: [RHEL7 PATCH 501/626] powerpc/boot: Add PROM_ERROR define in oflib
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 9cc36bb0aca622539be59266012d8be4346f0996
Author: Cedric Le Goater <clg@fr.ibm.com>
Date:   Thu Apr 24 09:23:29 2014 +0200

    powerpc/boot: Add PROM_ERROR define in oflib

    This is mostly useful to make to the boot wrapper code closer with
    the kernel code in prom_init.

    Signed-off-by: Cedric Le Goater <clg@fr.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/boot/of.h b/arch/powerpc/boot/of.h
index 40d95bf..0f058ef 100644
--- a/arch/powerpc/boot/of.h
+++ b/arch/powerpc/boot/of.h
@@ -23,4 +23,6 @@ typedef u32   __be32;
 #define cpu_to_be32(x) (x)
 #define be32_to_cpu(x) (x)
 
+#define PROM_ERROR (-1u)
+
 #endif /* _PPC_BOOT_OF_H_ */
diff --git a/arch/powerpc/boot/oflib.c b/arch/powerpc/boot/oflib.c
index 0f72b1a..7f61f2e 100644
--- a/arch/powerpc/boot/oflib.c
+++ b/arch/powerpc/boot/oflib.c
@@ -55,7 +55,7 @@ int of_call_prom(const char *service, int nargs, int nret, ...)
   args.args[nargs+i] = 0;
 
  if (prom(&args) < 0)
-  return -1;
+  return PROM_ERROR;
 
  return (nret > 0) ? be32_to_cpu(args.args[nargs]) : 0;
 }
@@ -80,9 +80,9 @@ static int of_call_prom_ret(const char *service, int nargs, int nret,
   args.args[nargs+i] = 0;
 
  if (prom(&args) < 0)
-  return -1;
+  return PROM_ERROR;
 
- if (rets != (void *) 0)
+ if (rets != NULL)
   for (i = 1; i < nret; ++i)
    rets[i-1] = be32_to_cpu(args.args[nargs+i]);
 
-- 
1.7.1